search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -149,46 +149,60 @@ class TemporalEntityHandler(
 }
 
 internal fun buildTemporalQuery(params: MultiValueMap<String, String>): TemporalQuery {
-    if (!params.containsKey("timerel") || !params.containsKey("time"))
-        throw BadRequestDataException("'timerel and 'time' request parameters are mandatory")
-
-    if (params.getFirst("timerel") == "between" && !params.containsKey("endTime"))
+    val timerelParam = params.getFirst("timerel")
+    val timeParam = params.getFirst("time")
+    val endTimeParam = params.getFirst("endTime")
+    val timeBucketParam = params.getFirst("timeBucket")
+    val aggregateParam = params.getFirst("aggregate")
+    val lastNParam = params.getFirst("lastN")
+    val attrsParam = params.getFirst("attrs")
+
+    if (timerelParam == "between" && endTimeParam == null)
         throw BadRequestDataException("'endTime' request parameter is mandatory if 'timerel' is 'between'")
 
-    val timerel = try {
-        TemporalQuery.Timerel.valueOf(params.getFirst("timerel")!!.toUpperCase())
-    } catch (e: IllegalArgumentException) {
-        throw BadRequestDataException("'timerel' is not valid, it should be one of 'before', 'between', or 'after'")
+    val (timerel, time) = if (timerelParam == null && timeParam == null) {
+        Pair(
+            TemporalQuery.Timerel.AFTER,
+            ("1970-01-01T00:00:00Z").parseTimeParameter("")
+        )
+    } else if (timerelParam != null && timeParam != null) {
+        val timeRelTemp: TemporalQuery.Timerel
+        try {
+            timeRelTemp = TemporalQuery.Timerel.valueOf(timerelParam.toUpperCase())
+        } catch (e: IllegalArgumentException) {
+            throw BadRequestDataException("'timerel' is not valid, it should be one of 'before', 'between', or 'after'")
+        }
+        Pair(timeRelTemp, timeParam.parseTimeParameter("'time' parameter is not a valid date"))
+    } else {
+        throw BadRequestDataException("'timerel' and 'time' must be both present")
```

Comment:
not exactly. maybe `'timerel' and 'time' must be both present if one of them is present`?

Reply:
it 's a matter of interpretation. If they are both present, one of them is present too. What about : 
timerel and time have to be used in conjunction ?